proxy/merging.go

```diff
@@ -82,82 +82,137 @@ func parallelMerge(timeout time.Duration, rc ResponseCombiner, next ...Proxy) Pr
-							for _, k := range keys[:len(keys)-1] {
-								v, ok = data[k]
-								if !ok {
-									break
-								}
-								switch clean := v.(type) {
-								case map[string]interface{}:
-									data = clean
-								default:
-									break
+
+		for i := range next {
+			go func(i int) {
+				n := next[i]
+				for _, j := range deps[i] {
+					select{
+					case <-done[j]:
+					case <-localCtx.Done():
+						return
+					}
+				}
+				var req *Request = request
+				if i > 0 {
+					req = CloneRequest(request)
+					for _, match := range matches[i] {
+						if len(match) > 1 {
+							rNum, err := strconv.Atoi(match[1])
+							if err != nil || rNum >= i || parts[rNum] == nil {
+								continue
+							}
+							key := "Resp" + match[1] + "_" + match[2]
+
+							var v interface{}
+							var ok bool
+
+							data := parts[rNum].Data
+							keys := strings.Split(match[2], ".")
+							if len(keys) > 1 {
+								for _, k := range keys[:len(keys)-1] {
+									v, ok = data[k]
+									if !ok {
+										break
+									}
+									switch clean := v.(type) {
+									case map[string]interface{}:
+										data = clean
+									default:
+										break
+									}
 								}
 							}
-						}
 
-						v, ok = data[keys[len(keys)-1]]
-						if !ok {
-							continue
-						}
-						switch clean := v.(type) {
-						case string:
-							request.Params[key] = clean
-						case int:
-							request.Params[key] = strconv.Itoa(clean)
-						case float64:
-							request.Params[key] = strconv.FormatFloat(clean, 'E', -1, 32)
-						case bool:
-							request.Params[key] = strconv.FormatBool(clean)
-						default:
-							request.Params[key] = fmt.Sprintf("%v", v)
+							v, ok = data[keys[len(keys)-1]]
+							if !ok {
+								continue
+							}
+							switch clean := v.(type) {
+							case string:
+								req.Params[key] = clean
+							case int:
+								req.Params[key] = strconv.Itoa(clean)
+							case float64:
+								req.Params[key] = strconv.FormatFloat(clean, 'E', -1, 32)
+							case bool:
+								req.Params[key] = strconv.FormatBool(clean)
+							default:
+								req.Params[key] = fmt.Sprintf("%v", v)
+							}
 						}
 					}
 				}
-			}
-			requestPart(localCtx, n, request, out, errCh)
-			select {
-			case err := <-errCh:
-				if i == 0 {
-					cancel()
-					return nil, err
+				requestPart(localCtx, n, req, out, errCh)
+				select {
+				case err := <-errCh:
+					ch <- partResult{i, err}
+				case response := <-out:
+					if !response.IsComplete {
+						cancel()
+						return
+					}
+					parts[i] = response
+					close(done[i])
+					ch <- partResult{i, nil}
 				}
-				acc.Merge(nil, err)
-				break TxLoop
-			case response := <-out:
-				acc.Merge(response, nil)
-				if !response.IsComplete {
-					break TxLoop
+			}(i)
+		}
+
+		for _ = range next {
+			select {
+			case res := <-ch:
+				if i, err := res.i, res.err; err != nil {
+					if i == 0 {
+						cancel()
+						return nil, err
+					}
```

Comment:
I'm not sure I got it right, but at this point isn't all the requests already made?

If that is true, then I think we can skip the check `i == 0` to `cancel()` and `return`

Reply:
You are right, I think I did this to work around the existing tests; before this change, everything was executed in the order, so it didn't make sense to proceed after the first request failed.
I will see if I can infer the right behavior here.

Reply:
Got it, I couldn't see anything else so it looks good to me

Reply:
I could only find out that we can just halt the sequence if any required call failed, in order to keep the tests passing. I think it's a fair behavior.
Updated.